tools/xcodeproj_shims/build-wrapper.sh

```diff
@@ -1,15 +1,21 @@
 #!/bin/bash
 set -euxo pipefail
 
-EXEC_LOG_OPTION=""
+BAZEL_BUILD_OPTIONS="--build_event_text_file=$BAZEL_BUILD_EVENT_TEXT_FILENAME"
+BAZEL_BUILD_OPTIONS+=" --build_event_publish_all_actions"
+
 if [ $BAZEL_EXECUTION_LOG_ENABLED -gt 0 ]; then
-    EXEC_LOG_OPTION="--experimental_execution_log_file=$BAZEL_BUILD_EXECUTION_LOG_FILENAME"
+    BAZEL_BUILD_OPTIONS+=" --experimental_execution_log_file=$BAZEL_BUILD_EXECUTION_LOG_FILENAME"
+fi
+
+# When building for an iOS device in XCode, TARGET_DEVICE_IDENTIFIER is exported, and PLATFORM_NAME is iphoneos.
+if [ -n "$(export -p | grep TARGET_DEVICE_IDENTIFIER=)" ] && [ "$PLATFORM_NAME" = "iphoneos" ]; then
+    BAZEL_BUILD_OPTIONS+=" --ios_multi_cpus=arm64"
```

Comment:
should we echo `using arm64 since detected abc` ad a way to tell the scirpt is altering build bahavior/config?

Reply:
Why are you not able to do `if [ -n ${TARGET_DEVICE_IDENTIFIER:-} ] && [ "$PLATFORM_NAME" = "iphoneos" ]; then`

Reply:
should we maybe instead check the archs env vars? Because I think building for a "generic" device is a thing?

Reply:
PLATFORM_NAME is `iphoneos` when building for generic device.